data-simulation/sessionData.js

```diff
@@ -128,23 +133,51 @@ const generateSessions = (date, days) => {
   return writeSequentially(0);
 };
 
-// Generates user session data
-// Makes HTTP post request to /sessions every 1 second
-const simulateLiveData = () => {
-  cron.schedule('0-59 * * * * *', () => {
-    const options = {
-      uri: 'http://localhost:3000/sessions',
-      method: 'POST',
-      json: new Session(new Date()),
-    };
-    console.log(options.json.events);
-    request(options, (err, res, body) => {
-      if (err) {
-        console.log(err);
-      }
-      console.log(body);
-    });
-  }, true);
+// DEPRECATED, use AWS SQS instead
+// Generates one user session data
+// Makes HTTP post request to /sessions
+// (will be replaced once AWS SQS is implemented)
+const sendSessionDataHTTP = () => {
+  const options = {
+    uri: 'http://localhost:3000/sessions',
+    method: 'POST',
+    json: new Session(new Date()),
+  };
+  request(options, (err, res, body) => {
+    if (err) {
+      console.log(err);
+    }
+    console.log(body);
+  });
 };
 
-module.exports = { simulateLiveData, generateSessions };
+// DEPRECATED, use AWS SQS instead
+// Simulate live session data flow via HTTP post request every 1 second
+const simulateLiveDataHTTP = () => cron.schedule('0-59 * * * * *', sendSessionDataHTTP, true);
+
+// Generates one user session data
+// Send simulated sesssion data (input) to AWS SQS
+const sendSessionDataSQS = () => {
+  const sqs = new AWS.SQS();
```

Comment:
It looks like instantiating a new sqs class every time you invoke this. I think you can have the same functionality without the added memory cost if you kept sqs in the outer scope. Let me know what you think!

Reply:
You're right, I am instantiating it every second inside my cron schedule function!  Thanks for pointing it out!